wgpu-core/src/command/bundle.rs

```diff
@@ -656,17 +656,45 @@ impl State {
 
 /// Error encountered when finishing recording a render bundle.
 #[derive(Clone, Debug, Error)]
-pub enum RenderBundleError {
+pub enum RenderBundleErrorInner {
     #[error(transparent)]
     Device(#[from] DeviceError),
     #[error(transparent)]
-    RenderCommand(#[from] RenderCommandError),
+    RenderCommand(RenderCommandError),
     #[error(transparent)]
     ResourceUsageConflict(#[from] UsageConflict),
     #[error(transparent)]
     Draw(#[from] DrawError),
 }
 
+impl<T> From<T> for RenderBundleErrorInner
+where
+    T: Into<RenderCommandError>,
+{
+    fn from(t: T) -> Self {
+        Self::RenderCommand(t.into())
+    }
+}
+
+/// Error encountered when finishing recording a render bundle.
+#[derive(Clone, Debug, Error)]
+#[error("{command}")]
+pub struct RenderBundleError {
+    command: &'static str,
+    #[source]
+    inner: RenderBundleErrorInner,
+}
+
+fn pass_err<E>(command: &'static str) -> impl FnOnce(E) -> RenderBundleError
```

Comment:
would we be able to copy the result if it's `impl Fn(E) -> ...` instead of `FnOnce`?
That would allow us to do `let pass_ctx = pass_err("In a bundle parameter");` right away and avoid repeating the `pass_err` calls through arms of the match

Reply:
I have thought about that, but that doesn't actually solve the issue because different uses of this function want to instantiate the closure with different `E` types.

Reply:
With the `map_pass_err`, this is a moot point now.